### ISSUE TITLE
DuplicatedArrayKey rule  is not detecting errors outside the function (return array type files)

### ISSUE THREAD

**Issue by glensc** [phpmd/phpmd#1215]
- PHPMD version: 2.15.0
- PHP Version: 8.2.24
- Installation type: composer
- Operating System / Distribution & Version: macOS Sonoma 14.7 

### Current Behavior

Second problem from https://github.com/phpmd/phpmd/issues/1214 issue

The `DuplicatedArrayKey` rule  is not detecting errors outside the function.

### Expected Behavior

Report the DuplicatedArrayKey violation

### Steps To Reproduce:

```xml
# cat phpmd.xml
<?xml version="1.0"?>
<ruleset name="Custom Ruleset for DuplicatedArrayKey"
         xmlns="https://pmd.github.io/pmd-5.0.0"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="https://pmd.github.io/pmd-5.0.0 https://pmd.github.io/pmd-5.0.0.xsd"
         xsi:noNamespaceSchemaLocation="https://pmd.github.io/pmd-5.0.0.xsd">

    <description>
        This ruleset applies only the DuplicatedArrayKey rule.
    </description>

    <!-- Define the DuplicatedArrayKey rule -->
    <rule name="DuplicatedArrayKey"
          message="The array contains duplicated keys, which can lead to unexpected behavior."
          class="PHPMD\Rule\CleanCode\DuplicatedArrayKey">
        <description>
            This rule detects duplicated keys in arrays.
        </description>
    </rule>
</ruleset>
```

```php
# cat config/bp.php
<?php

return [
    'b2b_bank_name' => 'AS Swedbank',
    "b2b_bank_name" => 'AS Swedbank',
];
```

```
$ phpmd config/bp.php ansi phpmd.xml

Found 0 violations and 0 errors in 136ms

No mess detected
```

NOTE, if the same code is wrapped in function, it will detect theerror:
- https://github.com/phpmd/phpmd/issues/1214#issuecomment-2422175186

## Checks before submitting
* [x] Be sure that there isn't already an issue about this. See: [Issues list](https://github.com/phpmd/phpmd/issues)
* [x] Be sure that there isn't already a pull request about this. See: [Pull requests](https://github.com/phpmd/phpmd/pulls)
* [x] I have added every step to reproduce the bug.
* [x] If possible I added relevant code examples.
* [x] This issue is about 1 bug and nothing more.
* [x] The issue has a descriptive title. For example:  "JSON rendering failed on Windows for filenames with space".

**Comment by Morgy93** [comment]
Hello,

it's not the `DuplicatedArrayKey` rule. In fact, the issue is that code **outside of functions** is not being checked at all.

For example, this code:

```php
<?php

$a = true;
```

is **not** flagged as a problem, whereas this code:

```php
<?php

function b() {
    $a = true;
}
```

**is** detected as a mess.

PHPMD 2.15.0

**Comment by AJenbo** [comment]
Duplicate of https://github.com/phpmd/phpmd/issues/681
